lib/src/from_handlers.dart

```diff
@@ -67,12 +67,9 @@ class _StreamTransformer<S, T> implements StreamTransformer<S, T> {
       controller.onResume = () => subscription?.resume();
     }
     controller.onCancel = () {
-      if (controller.hasListener || subscription == null) {
-        return new Future.value();
-      }
-      var toCancel = subscription;
+      if (controller.hasListener || subscription == null) return;
+      subscription.cancel();
```

Comment:
If I forward the returned Future from `subscription.cancel()` then there is an extra microtask between the source stream finishing it's `.close()` and the result Stream calling it's subscribers `onDone`.

Ideally I'd like `onDone` behavior consistent with the source stream. I can get that with a sync broadcast stream, or by ignoring the Future here - but I don't get it with a single-subscriber stream and returning this Future.

See this in action here: https://dartpad.dartlang.org/35319efd7419556622f76c9cc124b9d7

If this is an unreasonable goal - if the onDone callbacks are not meant to complete within the Future returned by the source controller `.close()` - then I suppose I could add back some extra `await`s in the tests...